Realm/Realm.Weaver/Analytics/Analytics.cs

```diff
@@ -0,0 +1,884 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+using static RealmWeaver.AnalyticsUtils;
+using static RealmWeaver.Metric.SdkFeature;
+
+namespace RealmWeaver
+{
+    // Asynchronously submits build information to Realm when the assembly weaver
+    // is running
+    //
+    // To be clear: this does *not* run when your app is in production or on
+    // your end-user's devices; it will only run when you build your app from source.
+    //
+    // Why are we doing this? Because it helps us build a better product for you.
+    // None of the data personally identifies you, your employer or your app, but it
+    // *will* help us understand what Realm version you use, what host OS you use,
+    // etc. Having this info will help with prioritizing our time, adding new
+    // features and deprecating old ones. Collecting an anonymized assembly name &
+    // anonymized MAC is the only way for us to count actual usage of the other
+    // metrics accurately. If we don't have a way to deduplicate the info reported,
+    // it will be useless, as a single developer building their app on Windows ten
+    // times would report 10 times more than a single developer that only builds
+    // once from Mac OS X, making the data all but useless. No one likes sharing
+    // data unless it's necessary, we get it, and we've debated adding this for a
+    // long long time. Since Realm is a free product without an email sign-up, we
+    // feel this is a necessary step so we can collect relevant data to build a
+    // better product for you.
+    //
+    // Currently the following information is reported:
+    // - What OS and CPU architecture you are running on
+    // - What OS and CPU architecture you are building for
+    // - What version of the Realm SDK you're using
+    // - What framework and what framework version Realm is being used with (e.g. Xamarin, MAUI, etc.)
+    // - How the Realm SDK was installed (e.g. Nuget, manual, etc.)
+    // - What APIs of the Realm SDK you're using
+    // - An anonymized MAC address and assembly name ID to aggregate the other information on.
+    internal class Analytics
+    {
+        private readonly ImportedReferences _references;
+        private readonly ILogger _logger;
+
+        #region FeatureDictionaries
+
+        private readonly Dictionary<string, byte> _realmFeaturesToAnalyze = new Dictionary<string, byte>()
```

Comment:
Target-typed obeject creation is not available in C# 8.0. It's only from C# 9.0 onwards.
It seems that Unity [supports](https://docs.unity3d.com/Manual/CSharpCompiler.html) most of C# 9.0. If you want, later I can look into bumping up the version of the language in `Realm.Fody` and all the projects that involve unity.
Let me know, so that I know if to create an issue for that.

Reply:
We are compiling those assemblies using msbuild, so Unity support should not affect us.